cmake/Platform/Common/GCC/Configurations_gcc.cmake

```diff
@@ -73,6 +74,12 @@ ly_append_configurations_options(
         -Wno-switch
         -Wno-uninitialized
         -Wno-unused-result
+        -Wno-dangling-pointer
+        -Wno-error=unused-but-set-variable
+        -Wno-error=unused-variable
+        -Wno-error=subobject-linkage
+        -Wno-error=nonnull
+        -Wno-error=free-nonheap-object
```

Comment:
free non-heap
does this really happen or it's a false positive?

Reply:
about -Wfree-nonheap-object:
```
In destructor ‘_i_reference_target<Counter>::~_i_reference_target() [with Counter = int]’,
    inlined from ‘void _i_reference_target<Counter>::Release() [with Counter = int]’ at /media/slapin/data/o3de/Code/Legacy/CryCommon/./smartptr.h:335:13,
    inlined from ‘_smart_ptr<_I>::~_smart_ptr() [with _I = CUiAnimViewKeyUIControls]’ at /media/slapin/data/o3de/Code/Legacy/CryCommon/./smartptr.h:67:23,
    inlined from ‘constexpr void std::_Destroy(_Tp*) [with _Tp = _smart_ptr<CUiAnimViewKeyUIControls>]’ at /usr/include/c++/13/bits/stl_construct.h:151:22,
    inlined from ‘static void std::_Destroy_aux<<anonymous> >::__destroy(_ForwardIterator, _ForwardIterator) [with _ForwardIterator = _smart_ptr<CUiAnimViewKeyUIControls>*; bool <anonymous> = false]’ at /usr/include/c++/13/bits/stl_construct.h:163:19,
    inlined from ‘void std::_Destroy(_ForwardIterator, _ForwardIterator) [with _ForwardIterator = _smart_ptr<CUiAnimViewKeyUIControls>*]’ at /usr/include/c++/13/bits/stl_construct.h:196:11,
    inlined from ‘void std::_Destroy(_ForwardIterator, _ForwardIterator, allocator<_T2>&) [with _ForwardIterator = _smart_ptr<CUiAnimViewKeyUIControls>*; _Tp = _smart_ptr<CUiAnimViewKeyUIControls>]’ at /usr/include/c++/13/bits/alloc_traits.h:947:15,
    inlined from ‘std::vector<_Tp, _Alloc>::~vector() [with _Tp = _smart_ptr<CUiAnimViewKeyUIControls>; _Alloc = std::allocator<_smart_ptr<CUiAnimViewKeyUIControls> >]’ at /usr/include/c++/13/bits/stl_vector.h:732:15,
    inlined from ‘virtual CUiAnimViewKeyPropertiesDlg::~CUiAnimViewKeyPropertiesDlg()’ at /media/slapin/data/o3de/Gems/LyShine/Code/Editor/Animation/UiAnimViewKeyPropertiesDlg.h:147:7,
    inlined from ‘virtual CUiAnimViewKeyPropertiesDlg::~CUiAnimViewKeyPropertiesDlg()’ at /media/slapin/data/o3de/Gems/LyShine/Code/Editor/Animation/UiAnimViewKeyPropertiesDlg.h:147:7:
/media/slapin/data/o3de/Code/Legacy/CryCommon/./smartptr.h:324:5: error: ‘void operator delete(void*, std::size_t)’ called on pointer ‘*(CUiAnimViewKeyUIControls**)((char*)__first)’ with nonzero offset 16 [-Werror=free-nonheap-object]
  324 |     }
      |     ^
In destructor ‘_i_reference_target<Counter>::~_i_reference_target() [with Counter = int]’,
    inlined from ‘void _i_reference_target<Counter>::Release() [with Counter = int]’ at /media/slapin/data/o3de/Code/Legacy/CryCommon/./smartptr.h:335:13,
    inlined from ‘_smart_ptr<_I>::~_smart_ptr() [with _I = CUiAnimViewKeyUIControls]’ at /media/slapin/data/o3de/Code/Legacy/CryCommon/./smartptr.h:67:23,
    inlined from ‘constexpr void std::_Destroy(_Tp*) [with _Tp = _smart_ptr<CUiAnimViewKeyUIControls>]’ at /usr/include/c++/13/bits/stl_construct.h:151:22,
    inlined from ‘static void std::_Destroy_aux<<anonymous> >::__destroy(_ForwardIterator, _ForwardIterator) [with _ForwardIterator = _smart_ptr<CUiAnimViewKeyUIControls>*; bool <anonymous> = false]’ at /usr/include/c++/13/bits/stl_construct.h:163:19,
    inlined from ‘void std::_Destroy(_ForwardIterator, _ForwardIterator) [with _ForwardIterator = _smart_ptr<CUiAnimViewKeyUIControls>*]’ at /usr/include/c++/13/bits/stl_construct.h:196:11,
    inlined from ‘void std::_Destroy(_ForwardIterator, _ForwardIterator, allocator<_T2>&) [with _ForwardIterator = _smart_ptr<CUiAnimViewKeyUIControls>*; _Tp = _smart_ptr<CUiAnimViewKeyUIControls>]’ at /usr/include/c++/13/bits/alloc_traits.h:947:15,
    inlined from ‘void std::vector<_Tp, _Alloc>::_M_realloc_insert(iterator, _Args&& ...) [with _Args = {_smart_ptr<CUiAnimViewKeyUIControls>}; _Tp = _smart_ptr<CUiAnimViewKeyUIControls>; _Alloc = std::allocator<_smart_ptr<CUiAnimViewKeyUIControls> >]’ at /usr/include/c++/13/bits/vector.tcc:517:15:
/media/slapin/data/o3de/Code/Legacy/CryCommon/./smartptr.h:324:5: error: ‘void operator delete(void*, std::size_t)’ called on pointer ‘*(CUiAnimViewKeyUIControls**)((char*)__first)’ with nonzero offset 16 [-Werror=free-nonheap-object]
  324 |     }
      |     ^
In destructor ‘_i_reference_target<Counter>::~_i_reference_target() [with Counter = int]’,
    inlined from ‘void _i_reference_target<Counter>::Release() [with Counter = int]’ at /media/slapin/data/o3de/Code/Legacy/CryCommon/./smartptr.h:335:13,
    inlined from ‘_smart_ptr<_I>::~_smart_ptr() [with _I = CUiAnimViewKeyUIControls]’ at /media/slapin/data/o3de/Code/Legacy/CryCommon/./smartptr.h:67:23,
    inlined from ‘CUiAnimViewKeyPropertiesDlg::CUiAnimViewKeyPropertiesDlg(QWidget*)’ at /media/slapin/data/o3de/Gems/LyShine/Code/Editor/Animation/UiAnimViewKeyPropertiesDlg.cpp:89:28:
/media/slapin/data/o3de/Code/Legacy/CryCommon/./smartptr.h:324:5: error: ‘void operator delete(void*, std::size_t)’ called on pointer ‘<unknown>’ with nonzero offset 16 [-Werror=free-nonheap-object]
  324 |     }
      |     ^
In file included from /media/slapin/data/o3de-projects/pong/build/linux/External/LyShine-dae488a6/Code/CMakeFiles/LyShine.Tools.Static.dir/Unity/unity_14_cxx.cxx:7:
/media/slapin/data/o3de/Gems/LyShine/Code/Editor/Animation/UiAnimViewKeyPropertiesDlg.cpp: In constructor ‘CUiAnimViewKeyPropertiesDlg::CUiAnimViewKeyPropertiesDlg(QWidget*)’:
/media/slapin/data/o3de/Gems/LyShine/Code/Editor/Animation/UiAnimViewKeyPropertiesDlg.cpp:87:65: note: returned from ‘void* operator new(std::size_t)’
   87 |     CUiAnimViewTrackEventKeyUIControls* trackEventControl = new CUiAnimViewTrackEventKeyUIControls;
      |                                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
I dont see how the `CUiAnimViewKeyUIControls` class needs to be derived from QObject nor inherit QObject in its class.  Might be best to remove that and see if the alignment cleans up - all other ref counted _smart_ptr objects are single-inheritors with no offset... ?

one issue is I can't seem to repro these, on clang16 or clang14 with asan / fortify.  So I can't experiment with different options to see if it can be fixed.

Reply:
Yeah, I tried removing the QObject inherit and Q_OBJECT in its class def from CUiAnimViewKeyUIControls and the track view seems to work just fine and show those key ui controls without issue, I think it was an oversight.  Perhaps removing it will clear the issue for you

Reply:
Cool, indeed it fixed the issue with -Wfree-nonheap-object, Thanks a lot!